mtls_certificates.go

```diff
@@ -0,0 +1,135 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// MTLSAssociationDetails represents the metadata for an existing association between a user-uploaded mTLS certificate and a Cloudflare service.
+type MTLSAssociationDetails struct {
+	Service string `json:"service"`
+	Status  string `json:"status"`
+}
+
+// MTLSAssociationResponse represents the response from the retrieval endpoint for mTLS certificate associations.
+type MTLSAssociationResponse struct {
+	Response
+	Result []MTLSAssociationDetails `json:"result"`
+}
+
+// MTLSCertificateDetails represents the metadata for a user-uploaded mTLS certificate.
+type MTLSCertificateDetails struct {
+	ID           string    `json:"id"`
+	Name         string    `json:"name"`
+	Issuer       string    `json:"issuer"`
+	Signature    string    `json:"signature"`
+	SerialNumber string    `json:"serial_number"`
+	Certificates string    `json:"certificates"`
+	CA           bool      `json:"ca"`
+	UploadedOn   time.Time `json:"uploaded_on"`
+	UpdatedAt    time.Time `json:"updated_at"`
+	ExpiresOn    time.Time `json:"expires_on"`
+}
+
+// MTLSCertificateResponse represents the response from endpoints relating to retrieving, creating, and deleting an mTLS certificate.
+type MTLSCertificateResponse struct {
+	Response
+	Result MTLSCertificateDetails `json:"result"`
+}
+
+// MTLSCertificatesResponse represents the response from the mTLS certificate list endpoint.
+type MTLSCertificatesResponse struct {
+	Response
+	Result []MTLSCertificateDetails `json:"result"`
+}
+
+// MTLSCertificateParams represents the data related to the mTLS certificate being uploaded. Name is an optional field.
+type MTLSCertificateParams struct {
+	Name         string `json:"name"`
+	Certificates string `json:"certificates"`
+	PrivateKey   string `json:"private_key"`
+	CA           bool   `json:"ca"`
+}
+
+// ListMTLSCertificates returns a list of all user-uploaded mTLS certificates.
+//
+// API reference: https://api.cloudflare.com/#mtls-certificate-management-list-mtls-certificates
+func (api *API) ListMTLSCertificates(ctx context.Context, accountID string) ([]MTLSCertificateDetails, error) {
```

Comment:
let's update all the methods to match the signature from https://github.com/cloudflare/cloudflare-go/blob/master/docs/experimental.md#consistent-crud-method-signatures. check out #1149 of a recent PR where this happened

Reply:
the list method should automatically paginate all results - like https://github.com/cloudflare/cloudflare-go/blob/master/filter.go#L113-L145

Reply:
Okay, I believe I've updated my functions to reflect this style. The only one that doesn't work with it is `ListMTLSCertificateAssociations` - the API does not allow pagination and does not return any result information.

Reply:
that's cool. let's chuck it in for future proofing (the original intent of the third parameter) and it can be passed as an empty.

i'll review this now.